python/paddle/distributed/auto_parallel/static/utils.py

```diff
@@ -2373,3 +2373,66 @@ def __impl__(*args, **kwargs):
 
 
 dygraph_guard = wrap_decorator(_dygraph_guard_)
+
+
+def wrap_data_for_completion(
+    dist_op, input_names: list, output_names: list, attr_names: list
+):
+    """
+    Get data used in inferring distributed attributes, including:
+      1. DistTensorSpec for each input and output tensor of this dist_op.
+      2. Operator attributes of this dist_op, e.g. transpose_x in matmul op.
+
+    Args:
+      dist_op: the DistributedOperator
+      input_names: list, name of the dist_op's input tensors
+      output_names: list, name of the dist_op's output tensors
+      attr_names: list, attribute name of the dist_op's corresponding serial op
+
+    Returns:
+      input_specs: list, DistTensorSpec for each input tensor of the dist_op
+      output_specs: list, DistTensorSpec for each output tensor of the dist_op
+      attrs: dict, attribute map of the dist op
+
+    Usage:
+      op_desc = dist_op.serial_op.desc
+      input_name_list = []
+      output_name_list = []
+      input_name_list.append(op_desc.input('X')[0]) # 'X' is the arg name for op
```

Comment:
it would be better (in future) that the wrap could take op as the only argument and user not need to border the input_name_list/output_name_list construction.

in order to achieve that wrap need to maintain the order of op argument slot from Phi API.

Reply:
The names of op arguments are inconsistent in static mode and Phi API, so it's too hard to obtain the arguments from op in wrap function.